src/common/cockpitfdpassing.c

```diff
@@ -119,6 +122,15 @@ cockpit_socket_receive_fd (int  socket_fd,
                         .msg_controllen = CMSG_LEN (sizeof *out_fd) };
   assert (msg.msg_controllen <= sizeof cmsg);
 
+  /* recvmsg() has no MSG_DONTWAIT, and e.g. sudo makes stdin non-blocking with `log_output` option */
+  if (fcntl (socket_fd, F_GETFL) & O_NONBLOCK)
+    {
+      struct pollfd socket_pfd = { .fd = socket_fd, .events = POLLIN };
```

Comment:
Any reason to not just do the `poll()` unconditionally?

Reply:
Not a strong one. My initial instinct was to add a `g_message()` about it, then noticed that this file doesn't use glib, didn't like a raw printf(), and then just "meh, not important enough" :grin: But if we treat it as a permanent hack for the C bridge (whose days are numbered anyway), I'm fine with dropping the `if`.

Reply:
Changed.